src/scicat_dataset.py

```diff
@@ -51,7 +55,7 @@ def convert_to_type(input_value: Any, dtype_desc: str) -> Any:
     return converter(input_value)
 
 
-def build_dataset_schema(
+def build_dataset_description(
```

Comment:
@YooSunYoung can we change the name of this function?
There is a field named description in dataset. I think it might be misleading.
How about **_build_dataset_instance_**?